app/services/dialog_local_service.rb

```diff
@@ -140,7 +101,7 @@ def determine_target_type(obj)
     when /^Service/
       "service"
     else
-      obj.class.name.demodulize.underscore
+      obj.class.base_model.name.underscore.downcase
```

Comment:
This seems like a much different return value in some cases.  Were you able to fully verify this change?

Reply:
The ones that are different from the old file with this change are: 
```cloud_network``` for what would have been ```private``` and
```host``` for what would've returned ```host_esx```